tests/scheduler/test_task_archivist.py

```diff
@@ -0,0 +1,251 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) GrimoireLab Contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+import json
+from unittest.mock import patch
+
+import rq
+
+from grimoirelab.core.scheduler.jobs import GrimoireLabJob
+from grimoirelab.core.scheduler.tasks.archivist import (
+    ArchivistProgress,
+    StorageBackend,
+    archivist_job
+)
+
+from ..base import GrimoireLabTestCase
+
+
+class MockStorageBackend(StorageBackend):
+    """Class to store events in the class itself for later inspection"""
+
+    events = []
+
+    def store(self, events: iter) -> int:
+        events = [e for e in events]
+        MockStorageBackend.events.extend(events)
+        return len(events)
+
+    @classmethod
+    def get_events(cls):
+        return cls.events
+
+    @classmethod
+    def clear_events(cls):
+        cls.events = []
+
+
+class TestArchivistJob(GrimoireLabTestCase):
+    """Unit tests for archivist_job function"""
+
+    def setUp(self):
+        MockStorageBackend.clear_events()
+        super().setUp()
+
+    def tearDown(self):
+        MockStorageBackend.clear_events()
+        super().tearDown()
+
+    def test_job(self):
+        """Test if events are stored correctly"""
+
+        # Add some sample events to the stream
+        events = [
```

Comment:
I would say to rename `events` to `expected_events`, so it's clear later to understand what's going on.